File: tcs_manager_system/src/main/java/com/tcsManager/tcsmanager/service/TestCaseService.java
package com.tcsManager.tcsmanager.service;

import java.util.List;
import java.util.Map;

import com.tcsManager.tcsmanager.entity.TestCase;
import com.tcsManager.tcsmanager.exception.TestCaseNotFoundException;
import com.tcsManager.tcsmanager.exception.UserNotFoundException;

import org.springframework.http.ResponseEntity;

/**
 * TestCaseService
 */
public interface TestCaseService {

    ResponseEntity<TestCase> getTestCase(long testCaseId) throws TestCaseNotFoundException;

    ResponseEntity<TestCase> createTestCase(TestCase testCase, long userId) throws UserNotFoundException;

    ResponseEntity<List<TestCase>> getAllTestCases();

    ResponseEntity<List<TestCase>> getTestCasesByUserId(long userId) throws UserNotFoundException;

    ResponseEntity<TestCase> updateTestCase(long testCaseId, TestCase testCaseUpdated) throws TestCaseNotFoundException;

    ResponseEntity<TestCase> partialUpdateTestCase(long testCaseId, Map<Object, Object> fields)
            throws TestCaseNotFoundException;

    ResponseEntity<?> deleteTestCase(long testCaseId) throws TestCaseNotFoundException;
}

File: tcs_manager_frontend/src/service/TestCaseService.js
import axios from "axios";

const TEST_CASE_BASE_API_URL = "http://localhost:8080/api/v1/test-cases/";

class TestCaseService {
    getAllTestCases() {
        return axios.get(TEST_CASE_BASE_API_URL)
            .then(response => response.data)
    }
    getTestCase(testCaseId) {
        return axios.get(TEST_CASE_BASE_API_URL + `${testCaseId}`)
            .then(response => response.data)
    }

    getTestCasesByUserId(userId) {
        return axios.get(TEST_CASE_BASE_API_URL + `u/${userId}`)
            .then(response => response.data)
    }

    createTestCase(userId, testCase) {
        return axios.post(TEST_CASE_BASE_API_URL + `u/${userId}`, testCase)
            .then(response => response.data)
    }

    updateTestCase(testCaseId) {
        return axios.put(TEST_CASE_BASE_API_URL + `${testCaseId}`)
            .then(response => response.data)
    }

    partialUpdateTestCase(testCaseId, testCase) {
        return axios.patch(TEST_CASE_BASE_API_URL + `${testCaseId}`, testCase)
            .then(response => response.data)
    }

    deleteTestCase(testCaseId) {
        return axios.delete(TEST_CASE_BASE_API_URL + `${testCaseId}`)
            .then(response => response.data)
    }
}

export default new TestCaseService();
